addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/handler/FreeboxHandler.java

```diff
@@ -186,80 +186,119 @@ public void initialize() {
+                        useHttps = true;
+                        fqdn = apiDomain + ":" + httpsPort;
+                    }
+                }
             }
 
-            if ((apiBaseUrl != null) && (apiVersion != null) && (hardwareVersion != null)) {
-                if (authorize(false, configuration.fqdn, apiBaseUrl, apiVersion)) {
-                    updateStatus(ThingStatus.ONLINE);
-
-                    if (globalJob == null || globalJob.isCancelled()) {
-                        long pollingInterval = getConfigAs(FreeboxServerConfiguration.class).refreshInterval;
-                        logger.debug("Scheduling server state update every {} seconds...", pollingInterval);
-                        globalJob = scheduler.scheduleAtFixedRate(globalRunnable, 1, pollingInterval,
-                                TimeUnit.SECONDS);
-                    }
+            if (errorMsg != null) {
+                logger.info("Bad thing configuration: {}", errorMsg);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            } else if (!authorize(useHttps, fqdn, apiBaseUrl, apiVersion)) {
+                if (StringUtils.isEmpty(configuration.appToken)) {
+                    errorMsg = "App token not set in the thing configuration";
                 } else {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                    errorMsg = "Check your app token in the thing configuration; opening session with " + fqdn
+                            + " using " + (useHttps ? "HTTPS" : "HTTP") + " API version " + apiVersion + " failed";
                 }
+                logger.info("{}", errorMsg);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
             } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                logger.info("Session opened with {} using {} API version {}", fqdn, (useHttps ? "HTTPS" : "HTTP"),
+                        apiVersion);
+                updateStatus(ThingStatus.ONLINE);
+
+                if (globalJob == null || globalJob.isCancelled()) {
+                    long pollingInterval = getConfigAs(FreeboxServerConfiguration.class).refreshInterval;
+                    logger.debug("Scheduling server state update every {} seconds...", pollingInterval);
+                    globalJob = scheduler.scheduleAtFixedRate(globalRunnable, 1, pollingInterval, TimeUnit.SECONDS);
+                }
             }
 
             Map<String, String> properties = editProperties();
             boolean update = false;
-            if ((apiBaseUrl != null) && !apiBaseUrl.isEmpty()
-                    && ((properties.get(FreeboxBindingConstants.API_BASE_URL) == null)
-                            || !properties.get(FreeboxBindingConstants.API_BASE_URL).equals(apiBaseUrl))) {
+            if (StringUtils.isNotEmpty(apiBaseUrl)
```

Comment:
What do you want to test here, the current version is indeed not null safe, sorry.